packages/menu/src/useMenu.ts

```diff
@@ -0,0 +1,751 @@
+      });
+
+      onChange({ type: changeType, isExpanded: false });
+    },
+    [onChange, isExpandedControlled]
+  );
+
+  const isItemSelected = useCallback(
+    (type, name, value) => {
+      switch (type) {
+        case 'checkbox': {
+          return !!controlledSelectedItems.find(item => item.value === value);
+        }
+        case 'radio': {
+          const match = controlledSelectedItems.filter(item => item.name === name)[0];
+
+          if (match) {
+            return match.value === value;
+          }
+
+          return false;
+        }
+        default: {
+          return undefined;
+        }
+      }
+    },
+    [controlledSelectedItems]
+  );
+
+  const getNextFocusedValue = useCallback(
+    ({ value, key, isAlphanumericChar }) => {
+      let nextFocusedValue = value;
+
+      if (isAlphanumericChar) {
+        const firstChars = menuItems.map(item =>
+          item.label ? item.label[0].toLowerCase() : String(item.value)[0].toLowerCase()
+        );
+        const index = firstChars.indexOf(key);
+        const item = menuItems[index];
+
+        if (item) {
+          nextFocusedValue = item.value;
+        }
+      } else {
+        const index = values.indexOf(value);
+        let nextIndex: number;
+
+        if (key === KEYS.UP) {
+          nextIndex = (index === 0 ? values.length : index) - 1;
+        } else if (key === KEYS.DOWN) {
+          nextIndex = (index === values.length - 1 ? -1 : index) + 1;
+        } else if (key === KEYS.END) {
+          nextIndex = values.length - 1;
+        } else if (key === KEYS.HOME) {
+          nextIndex = 0;
+        }
+
+        const item = menuItems[nextIndex!];
+
+        nextFocusedValue = item.value;
+      }
+
+      return nextFocusedValue;
+    },
+    [menuItems, values]
+  );
+
+  const getSelectedItems = useCallback(
+    ({ value, type, name, label, selected }) => {
+      let changes: ISelectedItem[] | null = [...controlledSelectedItems];
+
+      if (!type) return null;
+
+      const selectedItem = {
+        value,
+        type,
+        label,
+        ...(name && { name })
+      };
+
+      if (type === 'checkbox') {
+        if (selected) {
+          changes = changes.filter(item => item.value !== value);
+        } else {
+          changes.push(selectedItem);
+        }
+      } else if (type === 'radio') {
+        const index = changes.findIndex(item => item.name === name);
+
+        if (index > -1) {
+          changes.splice(index, 1, selectedItem);
+        } else {
+          changes.push(selectedItem);
+        }
+      }
+
+      return changes;
+    },
+    [controlledSelectedItems]
+  );
+
+  // Event
+
+  const handleTriggerClick = useCallback(
+    event => {
+      event.stopPropagation();
+
+      const changeType = StateChangeTypes.TriggerClick;
+
+      dispatch({
+        type: changeType,
+        payload: {
+          ...(!isFocusedValueControlled && { focusedValue: null }),
```

Comment:
Small thing to call attention to here. In the case where a menu is opened via trigger click, `focusedValue` is set to `null` intentionally to match the behavior outlined in [APG's submenu](https://www.w3.org/WAI/ARIA/apg/patterns/menubar/examples/menubar-navigation/#ex_label), where the click maintains focus on the trigger until an item is navigated to via mouse or keyboard.

<img width="406" alt="Screenshot 2023-08-22 at 2 44 59 PM" src="https://github.com/zendeskgarden/react-containers/assets/3946669/e546b6c6-f1ca-48b3-9878-4ce1be77a057">